### ISSUE TITLE
Fix bugs

### ISSUE THREAD

**Issue by HariWu1995** [TungPham1999/handwirte-digit-with-keras-new#1]
def load_dataset():
    # load dataset

    # split ra thành train và val
    (trainX, trainY), (testX, testY) = mnist.load_data()
    trainX = trainX.reshape((trainX.shape[0], 28, 28))
    trainX, valX, trainY, valY = train_test_split(trainX, trainY, test_size=0.25, random_state=42)

    # reshape dataset to have a single channel
    trainX = trainX.reshape((trainX.shape[0], 28, 28, 1))
    testX = testX.reshape((testX.shape[0], 28, 28, 1))
    valX = valX.reshape((valX.shape[0], 28, 28, 1))
    
    # one hot encode target values
    trainY = to_categorical(trainY)
    testY = to_categorical(testY)
    valY = to_categorical(valY)

    return trainX, trainY, testX, testY, valX, valY

def normalize(list_of_subsets):

    # convert from integers to floats
    for i, _ in enumerate(list_of_subsets):
        list_of_subsets[i] = list_of_subsets[i].astype('float32')
        list_of_subsets[i] = list_of_subsets[i] / 255.0

    return list_of_subsets

def run_test_harness():

    # load dataset
    trainX, trainY, testX, testY, valX, valY = load_dataset()

    # prepare pixel data
    [trainX, valX, testX] = normalize([trainX, valX, testX])

    # define model
    model = define_model()

    # fit model
    model.fit(trainX, trainY, 
              epochs=10, 
              batch_size=4, 
              verbose=0, 
              validation_data=(valX, valY))

    # save model
    model.save('final_model.h5')
    
    # load model
    loaded_model = load_model('final_model.h5')

    # evaluate model on test dataset
    _, acc = loaded_model.evaluate(testX, testY, verbose=0)
    print('> %.3f' % (acc * 100.0))
